src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfStates = [];
+  const stateCopy = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        const arrayToRemove = action.keysToRemove;
+
+        for (const key of arrayToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      case 'clear':
+        const arrayToDelete = Object.keys(stateCopy);
```

Comment:
```suggestion
        const keysToDelete = Object.keys(stateCopy);
```
It's better to specify what exactly array it is